src/localplayer.cpp

```diff
@@ -1048,3 +1035,48 @@ void LocalPlayer::old_move(f32 dtime, Environment *env, f32 pos_max_d,
 		m_can_jump = false;
 	}
 }
+
+float LocalPlayer::getSlipFactor(Environment *env, const v3f &speedH)
+{
+
+	if (!touching_ground)
+		return 1.0f;
+
+	float slip_factor = 1.0f;
+	// Slip on slippery nodes
+	const INodeDefManager *nodemgr = env->getGameDef()->ndef();
+	Map *map = &env->getMap();
+	const ContentFeatures &f = nodemgr->get(map->getNodeNoEx(
+			floatToInt(getPosition() - v3f(0, 0.05f * BS, 0), BS)));
+	int slippery = itemgroup_get(f.groups, "slippery");
+	if (!f.walkable) {
+		// leaning over an edge? Check surroundings for slippery nodes
+		slippery = 2 << 16; // guard value, bigger than all realistic ones
+		for (int z = 0; z <= 1; z++) {
+			for (int x = 0; x <= 1; x++) {
+				// this should cover all nodes surrounding player position
+				const ContentFeatures &f = nodemgr->get(map->getNodeNoEx(floatToInt(
+						getPosition() - v3f((x - 0.5f) * BS,
+										0.05f * BS,
```

Comment:
I just tried to rearrange it and ran clang-format on it again – back to the indentation from above. Should I just leave it like that?